src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -612,19 +791,105 @@ private SnapshotSummaryModel getSnapshotSummary(UUID snapshotId) {
+        new DRSObject()
+            // Extract singleton values
+            .id(extractUniqueDrsObjectValue(drsObjects, DRSObject::getId))
+            .name(extractUniqueDrsObjectValue(drsObjects, DRSObject::getName))
+            .description(extractUniqueDrsObjectValue(drsObjects, DRSObject::getDescription))
+            .size(extractUniqueDrsObjectValue(drsObjects, DRSObject::getSize))
+            .selfUri(extractUniqueDrsObjectValue(drsObjects, DRSObject::getSelfUri))
+            .mimeType(extractUniqueDrsObjectValue(drsObjects, DRSObject::getMimeType))
+            .version(extractUniqueDrsObjectValue(drsObjects, DRSObject::getVersion))
+            // Get the earliest creation date
+            .createdTime(
+                drsObjects.stream()
+                    .map(DRSObject::getCreatedTime)
+                    .map(Instant::parse)
+                    .min(Comparator.naturalOrder())
+                    .map(Instant::toString)
+                    .orElse(""))
+            // Get the latest update date
+            .updatedTime(
+                drsObjects.stream()
+                    .map(DRSObject::getUpdatedTime)
+                    .map(Instant::parse)
+                    .max(Comparator.naturalOrder())
+                    .map(Instant::toString)
+                    .orElse(""))
+            // Get a distinct list of access methods
+            .accessMethods(
+                drsObjects.stream()
+                    .map(DRSObject::getAccessMethods)
+                    .flatMap(Collection::stream)
+                    .distinct()
+                    .toList())
+            // Get a distinct list of checksums methods
+            .checksums(
+                drsObjects.stream()
+                    .map(DRSObject::getChecksums)
+                    .flatMap(Collection::stream)
+                    .distinct()
+                    .toList())
+            // Get a distinct list aliases
+            .aliases(
+                drsObjects.stream()
+                    .map(DRSObject::getAliases)
+                    .flatMap(Collection::stream)
+                    .distinct()
+                    .sorted()
+                    .toList())
+            // Since this only works for files, set contents to be null
+            .contents(null);
+
+    // If there are overlapping checksum types with different values, throw an error
+    Map<String, List<DRSChecksum>> checksumsByType =
+        drsObject.getChecksums().stream().collect(Collectors.groupingBy(DRSChecksum::getType));
+    checksumsByType.forEach(
+        (k, v) -> {
+          if (v.size() > 1) {
+            throw new InvalidDrsObjectException(
+                "Invalid DRS object. Many checksums for %s exist".formatted(k));
+          }
+        });
+    return drsObject;
+  }
+
+  /**
+   * Given a list of DRSObjects, extract a singleton value and fail if there are 0, 2 or more values
+   */
+  private static <R> R extractUniqueDrsObjectValue(
```

Comment:
Would you mind adding unit test(s) for this method as well?

Reply:
totally!

Reply:
(though incidentally, my thought was that this got exercised by some of the slightly bigger unit test)